Feature/develop interactivity refactor

maurofmferrao opened the pull request.

maurofmferrao committed:
Remove previous implementation from editor
relates to xibosignage/xibo#3491

maurofmferrao committed:
First Property Panel implementation (temp design)
relates to xibosignage/xibo#3491

maurofmferrao committed:
First Property Panel implementation (temp design)
relates to xibosignage/xibo#3491

maurofmferrao committed:
Actions: Next/Previous Layout shows regions as target (#2942)

relates to xibosignage/xibo#3621

maurofmferrao committed:
Fix for Action Type filter

maurofmferrao committed:
Remove elements in  group from trigger dropdown
relates to xibosignage/xibo#3491

maurofmferrao committed:
Action trigger dropdown: Highlight group and fix sort
relates to xibosignage/xibo#3491

maurofmferrao committed:
Fix Target and Trigger dropdown options
relates to xibosignage/xibo#3491

nadzpogi committed:
Add validation for add action form

maurofmferrao committed:
Interactivity: Arrow system - First version
relates to xibosignageltd/xibo-private#939

maurofmferrao committed:
Create Action by drag and drop - first version
relates to xibosignageltd/xibo-private#945

maurofmferrao committed:
Create Action by drag and drop - second version
relates to xibosignageltd/xibo-private#945

- Select Action object in Viewer added
- Add action from PP with selected trigger filled by default

maurofmferrao committed:
Interactivity: Layout dock
relates to xibosignageltd/xibo-private#949

maurofmferrao committed:
Interactivity: Layout dock - fixes
relates to xibosignageltd/xibo-private#949

 - Remove recent layouts from triggers
- Fix debounce call for search
- Add no results message to search
 - Clear recentes container when removing last one
- Remove hover to show scroll

maurofmferrao committed:
Interactivity: Layout dock - more tweaks
relates to xibosignageltd/xibo-private#949

 - Remove save to local storage
 - Remove name from Layout dock, show only code
 - Current Layout removed from layout search
 - Added to recents removed from layout search
 - Search dropdown styled as rest of editor
 - Escape shortcut to close search

maurofmferrao committed:
Action: Navigate to Layout
relates to xibosignageltd/xibo-private#951

 - makePagedSelect fix for default value when returning more than 1

maurofmferrao committed:
Merge branch 'develop' into feature/develop_interactivity_refactor

maurofmferrao committed:
Action: Navigate to Widget
relates to xibosignageltd/xibo-private#952

maurofmferrao committed:
Actions: Implement final UI
relates to xibosignageltd/xibo-private#958

dasgarner commented on lib/Factory/LayoutFactory.php:
```diff
@@ -2113,7 +2113,7 @@ public function getLayoutCodes($filterBy = []): array
         $parsedFilter = $this->getSanitizer($filterBy);
         $params = [];
-        $select = 'SELECT DISTINCT code, `campaign`.CampaignID, `campaign`.permissionsFolderId ';
+        $select = 'SELECT DISTINCT code, layout, `campaign`.CampaignID, `campaign`.permissionsFolderId ';
```
This needs a table prefix

dasgarner reviewed (changes requested):
Just one minor comment

maurofmferrao committed:
Get layout codes SQL fix
relates to xibosignage/xibo#3491

maurofmferrao committed:
Actions: Button widget
relates to xibosignageltd/xibo-private#957

maurofmferrao committed:
Interactive fixes
relates to xibosignage/xibo#3491

- Prevent adding more than one action for same trigger
- Navigate to Widget: Edit widget fixed
- Action toggle height corrected
- Action preview, names for next and precious actions added
- Action line update when resizing viewer

maurofmferrao committed:
Interactive fixes - part 2
relates to xibosignage/xibo#3491

- Add missing zone on edit for imported actions
- Action line trigger/target recalculate when refresh and not available
- Rogue lines removed
- Webhook action edit source fixed
 - Trigger code request loop fixed
 - Trigger removed from action view for webhook

maurofmferrao committed:
Interactive: Add Delete to Edit Action and buttons refactoring
relates to xibosignage/xibo#3491

maurofmferrao committed:
Interactive fixes - part 3
relates to xibosignage/xibo#3491

 - Prevent Delete via keyboard when editing Action Widget
 - Fix widget name not updating for Action edit
 - Update styling for drop-up menu for Action Widget menu

maurofmferrao committed:
Interactive fixes and tweaks - part 4
relates to xibosignage/xibo#3491

 - Highlight Action on viewer on PP hover
 - Published Layout: remove controls for Action Mode
 - Updated arrow creation to wait for target/trigger when loading
 - Next/Prev Layout target hidden and auto-populated

nadzpogi committed:
Validate widget actions before allowing layout publish

maurofmferrao committed:
Interactive fixes and tweaks - part 5
relates to xibosignage/xibo#3491

- Cancel button added to Widget from NavToWidget
 - Cancel on Edit reverts actions
 - Cancel on Add removes new widget

nadzpogi committed:
Validate actions from all levels before allowing layout publish

maurofmferrao committed:
Interactive fixes and tweaks - part 6
relates to xibosignage/xibo#3491

 - Fix for click to add for Action Widget
 - Fix for disabling Action Edit mode error

maurofmferrao committed:
Interactive fixes and tweaks - part 7
relates to xibosignage/xibo#3491

 - Edit action widget: Dark theme for drop area
- Cancel Add widget: No changes error fixed
- Main editor add by click not working after editing action widget
- Mini action helper highlight removal

maurofmferrao committed:
Interactive fixes and tweaks - part 8
relates to xibosignage/xibo#3491

 - Fixed widget not existing anymore ( still loads actions so they can be corrected )
 - Error grabbing widget and region name
 - Error for line highlight fixed

maurofmferrao committed:
Interactive fixes and tweaks - part 9
relates to xibosignage/xibo#3491

- Fix issue when cancelling a new Added Action widget, and still appearing on the Action Edit form
 - Fixed issue with action not being synced with BE in the property panel
- Fix harmless error ( by capturing it ) of setting start/end to lines

maurofmferrao committed:
Interactive fixes and tweaks - part 10
relates to xibosignage/xibo#3491

 - Fix for Cancel add action and showing wrong number of actions
 - Fix for new action showing Delete

maurofmferrao committed:
Add button static and scroll added to Action list
relates to xibosignage/xibo#3491

lykqa commented:
TEST: Layout/Editor END

dasgarner commented:
@maurofmferrao please can you carefully check through all of your commits which do not have a verified signature and post here if they are all you. I can bypass that rule and squash.

maurofmferrao commented:
> @maurofmferrao please can you carefully check through all of your commits which do not have a verified signature and post here if they are all you. I can bypass that rule and squash.

Not sure why those were not verified, but I can confirm all were committed by me. 👍🏻
